### ISSUE TITLE
Allow translator to enable detection for all translators in its frames

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero-connectors#156]
Interesting issue here:

https://forums.zotero.org/discussion/comment/281432/#Comment_281432

> The problem is that [5.0 disables translator detection on iframes](https://github.com/zotero/zotero/issues/1021) for performance reasons, and while we can set exceptions, the translator that actually needs to trigger in this case is the translator in the iframe (ScienceDirect in this case, but presumably that could be anything). But the way we've implemented this, the exception just triggers detection for that translator — in other words, it can cause the EBSCO translator's detection to run in all iframes on that site, but not the ScienceDirect translator's detection. We'll see if we can come up with a solution here. In the meantime, clicking to open the frame in a new tab is the best workaround.

So this would need some way to specify in the EBSCOhost translator that, when on resolver.ebscohost.com, detection should run for all translators. Not sure what a good solution is here. Maybe a function call in `detectWeb()`, like `detectAllTranslatorsInFrames()`?

Really we should probably just hard-code an exception for resolver.ebscohost.com in the connectors for now, and deal with this if another such site comes up.

**Comment by adomasven** [comment]
If we want to be more specific, I guess we could expose a function that would change the "root" frame of the doc and rerun Zotero detection. But I agree with a hard-coded exception for now.

**Comment by dstillman** [comment]
Yeah, that makes sense — it shouldn't ever be necessary to actually run in all frames.

**Comment by dstillman** [comment]
We should probably also support a custom meta tag for cases where the site admins expressly want this to work:

https://groups.google.com/d/topic/zotero-dev/gXshfoA24FM/discussion

We don't want to manage a large hard-coded list, and we don't want lots of one-line translators that just call the re-root function. (If we do add a function, we could actually implement the meta tag by just having the Embedded Metadata translator look for it and call the function.)

In the meantime, we should hard-code the site discussed there as well.

**Comment by adomasven** [comment]
Yeah, so this is definitely a solution for meta tags within a website, since a lot of these will be for generic translators like PICA or COinS

**Comment by dstillman** [comment]
I think we should probably also run detection automatically if there's only a single iframe, or always run it on the first frame found. I didn't check, but that alone might address all the examples given so far.

**Comment by adomasven** [comment]
Okay, so running on first frame was smart, but we now have a report from someone where the translation needs to occur on a second frame (sigh).

https://forums.zotero.org/discussion/69130/only-seeing-generic-translators-when-using-openurl-resolver-to-article

**Comment by adomasven** [comment]
How about a hidden whitelist pref? I wouldn't expect that everyone will be able to get hold of webadmins who can add meta tags, but with the first-frame inject solution this is such a fringe issue, that I think we can get away with a hidden pref.

CC @dstillman

**Comment by adam3smith** [comment]
Another idea would be to check for this setup specifically: The search.serialssolutions URL (as produced by the Summons discovery layer) is quite common, but it might be the only one that's doing this. We're using this here locally, too, so happy to help test.

**Comment by adomasven** [comment]
It's not, but we [previously had a temporary whitelist](https://github.com/zotero/zotero-connectors/commit/b0f0e83599bea997a0f9c99a093bab328f532973). We could bring that back too.

**Comment by dstillman** [comment]
I think we should still support a meta tag, but we should definitely just add the big ones we know of to the hard-coded whitelist. I don't think a client-side pref really makes sense — this should either be solved by the admin or by us when someone reports it. In the time it takes us to explain to someone how to fix this for themselves, we could just fix it for everyone.
